### PR TITLE
Update the README. Fixes grammar mistakes, adjusts examples to better…

### PR DESCRIPTION
**Pull request by gdborton**
… conform with guide, and removes non relevant pieces.

### CODE REVIEW

**Comment by gdborton** [comment]
@cmcculloh Hey, could you take a look at this?  Largely it is grammatical, and fixes a few issues where the examples don't conform with the style guide.

File: README.md [hunk 1]
```diff
@@ -166,16 +193,16 @@
...
 
 
-## <a name='functions'>Functions</a>
+## Functions
 
   - Function expressions:
 
     ```javascript
     // anonymous function as event handler
-    $el.on('click', function () {
```
Comment (cmcculloh) [inline README.md]: The space is required. Having a space between `function` and `()` helps to indicate that "function" is not the _name_ of the function, and that this is an un-named anonymous function.

File: README.md [hunk 2]
```diff
@@ -289,7 +317,10 @@
     var superPower = new SuperPower();
     ```
 
-  - Use multiple `var` declarations for multiple variables and declare each variable on a newline. Justification: http://benalman.com/news/2012/05/multiple-var-statements-javascript
+  - Use one `var` declaration per variable.
```
Comment (cmcculloh) [inline README.md]: Hrm... I like the direction here, but, if quickly scanning someone could easily stop after "declaration" and think you just use _one_ var per function. Maybe "Each variable needs it's own `var`" or "Use a `var` for each variable". Whatever it is, it needs to be a bit more scan-proof.

File: README.md [hunk 3]
```diff
@@ -522,37 +563,60 @@
...
+    if (test) return false;
+
     // good
     if (test) {
       return false;
     }
 
-    // good
-    function () { return false; }
+    // bad
```
Comment (cmcculloh) [inline README.md]: Why did this get changed to bad? Curly braces are optional at worst, and never to be considered bad.

File: README.md [hunk 4]
```diff
@@ -522,37 +563,60 @@
     if (test)
       return false;
 
+    // acceptable (only for very simple statements and early function return)
```
Comment (cmcculloh) [inline README.md]: I'm not at all a fan of this at all, and would change it if I saw it in any codebase I was in, but, I'm not going to fight against it here if it's super important to you.

File: README.md [hunk 5]
```diff
@@ -522,37 +563,60 @@
...
+  - Use `/** ... */` for multi-line comments. Include a description, specify types and values for all parameters and return values.
 
     ```javascript
     // bad
     // make() returns a new element
     // based on the passed in tag name
     //
-    // @param <String> tag
-    // @return <Element> element
+    // @param {String} tag
```
Comment (cmcculloh) [inline README.md]: Why the change to curly braces?

File: README.md [hunk 6]
```diff
@@ -606,29 +670,56 @@
     }
     ```
 
-    **[[⬆]](#TOC)**
+  - Prefixing your comments with `FIXME` or `TODO` helps other developers quickly understand if you're pointing out a problem that needs to be revisited, or if you're suggesting a solution to the problem that needs to be implemented. These are different than regular comments because they are actionable. The actions are `FIXME -- need to figure this out` or `TODO -- need to implement`.
```
Comment (cmcculloh) [inline README.md]: Using `--` here but `:` below... was that intentional?

File: README.md [hunk 7]
```diff
@@ -655,38 +746,63 @@
     });
     ```
 
-  - Place 1 space before anonymous function parenthesis.
+  - Place 1 space before the opening parenthesis in control statements (`if`, `while` etc.). Place no space before the argument list in function calls and declarations.
 
     ```javascript
-	// bad
```
Comment (cmcculloh) [inline README.md]: Why was this removed?

**Comment by cmcculloh** [comment]
Thanks @gdborton! I liked a lot of these changes, but had a few concerns.

File: README.md (hunk 5 above)
Comment (gdborton) [inline README.md]: This is to match the jsdoc's syntax, which I believe was the original intent.

http://usejsdoc.org/tags-param.html

File: README.md (hunk 3 above)
Comment (gdborton) [inline README.md]: It's not the curly braces, but the lack of the new line.  Using without a curly braces doesn't work with functions.

File: README.md (hunk 6 above)
Comment (gdborton) [inline README.md]: It wasn't exactly intentional, but either will work with most editor tools.  Did you have a strict preference for one over the other?

File: README.md (hunk 7 above)
Comment (gdborton) [inline README.md]: I removed for consistency's sake because that's what I'm used to.  Do you feel strongly about this?

**Comment by gdborton** [comment]
@cmcculloh Thanks for the feedback, I think I responded to everything.  It's awesome that you're so thorough.

File: README.md (hunk 6 above)
Comment (gdborton) [inline README.md]: Also, it isn't the syntax that's important so much as the comment prefixed with FIXME or TODO

File: README.md (hunk 3 above)
Comment (cmcculloh) [inline README.md]: lol, you are correct sir, using without curly braces most certainly does not work with functions. I didn't notice we had switched from `if` to `function`. OK, well, I can't even remember the last time I wrote a one line function that returned false, so, I guess I don't care either way.

File: README.md (hunk 5 above)
Comment (cmcculloh) [inline README.md]: OK cool, sounds good!

File: README.md (hunk 6 above)
Comment (cmcculloh) [inline README.md]: :+1:

File: README.md (hunk 7 above)
Comment (cmcculloh) [inline README.md]: I really like having a space between the keyword `function` and the `()`. It highlights the fact that you have not named the function and serves as a reminder to do so. The only thing that should be touching the parens is the name of the function itself.

I _could_ be persuaded by an argument that would accept line 763 as the "good" way for function definitions, as one may argue that `name ()` indicates instantiation whereas `name()` indicates execution. But, @tweettypography would have to come here and actually propose that officially ;)

Comment (tweettypography) [inline README.md]: If you think about ES6 method shorthand, then in my mind `fightIf (isJedi) { }` makes it much clearer that we are looking at a function definition. Does that help?

Comment (cmcculloh) [inline README.md]: Yeah, I totally get that. Like I said, if I received a pull request arguing for this as being our new accepted standard, I probably wouldn't vote against it.

File: README.md (hunk 2 above)
Comment (cmcculloh) [inline README.md]: :+1: Perfect! Thanks!

File: README.md [hunk 8]
```diff
@@ -655,38 +746,77 @@
...
-		var name;
-	}
+    // good
+    function fight() {
+      console.log('Swooosh!');
+    }
     ```
 
-  - Place an empty newline at the end of the file.
+  - Place 1 space before anonymous function parenthesis.
```
Comment (gdborton) [inline README.md]: @cmcculloh added this back in here.  Also, corrected anonymous functions throughout the readme.

**Comment by cmcculloh** [comment]
Awesome @gdborton, thanks! This all looks good to me. Thank you for your work! 

I'm going to give this until Monday for any of the other UXAs to weigh in before merging.

File: README.md [hunk 9]
```diff
@@ -412,7 +454,7 @@
 
       anonymous(); // => TypeError anonymous is not a function
 
-      var anonymous = function () {
+      var anonymous = function() {
```
Comment (cormacmccarthy) [inline README.md]: Oop! Sorry, just noticed you missed one here... Should be `var anonymous = function () {`

File: README.md [hunk 10]
```diff
@@ -853,27 +1046,27 @@
...
+    var o = {};
+    function c() {}
     var u = new user({
       name: 'Bob Parr'
     });
 
     // good
     var thisIsMyObject = {};
-    function thisIsMyFunction () {};
+    function thisIsMyFunction () {}
```
Comment (cormacmccarthy) [inline README.md]: Looks like a semicolon was accidentally removed here. Also, there's an errant space in `thisIsMyFunction()`

File: README.md [hunk 11]
```diff
@@ -993,19 +1186,19 @@
       this.set('lightsaber', lightsaber);
     }
 
-    Jedi.prototype.set = function (key, val) {
+    Jedi.prototype.set = function(key, val) {
```
Comment (cormacmccarthy) [inline README.md]: `function (key, val)`

File: README.md [hunk 12]
```diff
@@ -993,19 +1186,19 @@
...
       this[key] = val;
     };
 
-    Jedi.prototype.get = function (key) {
+    Jedi.prototype.get = function(key) {
```
Comment (cormacmccarthy) [inline README.md]: `function (key)`

File: README.md [hunk 13]
```diff
@@ -1039,26 +1232,26 @@
 
     ```javascript
     // bad
-    Jedi.prototype.jump = function () {
+    Jedi.prototype.jump = function() {
```
Comment (cormacmccarthy) [inline README.md]: I thought you indicated you went through and fixed all these, but, I keep seeing them. May want to do a global find/replace for `function(` with `function (`

**Comment by gdborton** [comment]
@cmcculloh Updated the anonymous functions to include a space before the paren (was some confusion on my part).  Also fix the missing semicolon.

File: README.md [hunk 14]
```diff
@@ -655,21 +746,55 @@
...
+
   - Place 1 space before anonymous function parenthesis.
 
     ```javascript
-	// bad
-	function() {
-		var name;
-	}
+    // bad
+    function () {
```
Comment (cormacmccarthy) [inline README.md]: oops, looks like we got an errant space in the "bad" example. Should probably be `function() {` for the "bad" example.

**Comment by cormacmccarthy** [comment]
One last thing and I'll merge. Thanks!

**Comment by gdborton** [comment]
Updated.

**Comment by cormacmccarthy** [comment]
Thank you sir!!!
